### PR TITLE
Added menu that allows opening folders, and viewing the images within.

### PR DESCRIPTION
**Pull request by heySanj**
Added a bit and did a little reorganising. 

- You can now select a folder to browse images within.
- Only '.png', '.jpg', '.jpeg', '.ico', '.gif', ~~'.svg'~~ files are displayed
- If there are no files to display, a blank screen is shown instead of the program crashing
- You can 'Refresh' a folder to see if new images are added/removed
- ~~Images are automatically resized to a height of 300px and a variable width (They keep their original aspect ratio)~~
- You can close the program using the Exit button 😃

### CODE REVIEW

**Comment by heySanj** [comment]
- Updated code to avoid the Bare Except
- Program no longer breaks if user doesn't select a folder to view
- Removed support for .svg files as tkinter couldn't display them
- Updated image resized so that only images larger than 500x300 are resized

💪

File: image-viewer.py
Comment (cBridges851) [inline image-viewer.py:11]: I'd say a lot of the comments that are in here aren't actually needed since the code is "self-documented", meaning what is done can easily be understood from the variable name or what the code is doing, even for beginners, so I'd suggest removing them :).

File: image-viewer.py
Comment (cBridges851) [inline image-viewer.py:12]: To relate to my previous comment, I guess this is one of those where you wouldn't necessarily know that "./images" is the default directory, so the comment helps, but there is an alternative:
```suggestion
        defaultDirectory = "./images"
        self.img_dir = "./images"
```

This makes it clear what "./images" is without needing a comment!

File: image-viewer.py
Comment (cBridges851) [inline image-viewer.py:30]: I'm probably missing something, but what is `and len(self.image_files) != 0` preventing? If there are 0 files, won't the `current_image_index` always be 0?

Comment (cBridges851) [reply image-viewer.py:30]: Likewise on line 35.

File: image-viewer.py
Comment (cBridges851) [inline image-viewer.py:67]: I had no idea what `tearoff` was prior to this, but I must say that it definitely looks better as False 😆

File: image-viewer.py
Comment (cBridges851) [inline image-viewer.py:83]: Really appreciate this addition!

File: image-viewer.py
Comment (cBridges851) [inline image-viewer.py:124]: This resize_image method is a really nice addition, it's much better than cropping the image like it was doing before - nice one!

File: image-viewer.py
Comment (cBridges851) [inline image-viewer.py:137]: Conditionals can also be laid out like:
```
if image.height > 300:
    ...
    return image.resize(...)

if image.height > 500:
    ...
    return image.resize(...)

return image
```

I don't think it will make any difference performance-wise, but it can help with readability. In this case though, I wouldn't say this is problematic in the slightest, so I'll leave it up to you whether you want to make that change or not!

File: image-viewer.py
Comment (cBridges851) [inline image-viewer.py:68]: Might be a job for another PR, depending on how big you think this would be, but perhaps there could be an option in the menu to Open File(s), so the user could navigate to a specific file via the file explorer and view it in the image viewer, or select multiple files in the file explorer and display in the same way as when you open a folder. That would save users having to create folders to put one or specific files in.

What are your thoughts?

**Review by cBridges851 — COMMENTED** [review]
I'm really happy with this PR! I loved having a play with this locally, I didn't come across any issues and I was delighted to see the new ability to select a folder of images, as well as the additional things of resizing, refreshing, exiting, and the considerations around if a folder is empty - you've clearly put lots of consideration into this, so a big thank you and well done!

I've put a few minor suggestions on this PR, I'll let you have a look at them and respond to them (feel free to disagree with anything by the way!), and then approve and merge after that. While going through, you'll see some resolved comments from me - that's just me pointing out things I'm super happy with 🌟 

Since this is the first PR in this repository and what you've done is awesome, I'd love to share this on my LinkedIn (https://www.linkedin.com/in/christa-bridges/). Would you be happy for me to do that, and have an account so I can tag you?

Many thanks and well done once again!

**Comment by heySanj** [comment]
> I'm really happy with this PR! I loved having a play with this locally, I didn't come across any issues and I was delighted to see the new ability to select a folder of images, as well as the additional things of resizing, refreshing, exiting, and the considerations around if a folder is empty - you've clearly put lots of consideration into this, so a big thank you and well done!
> 
> I've put a few minor suggestions on this PR, I'll let you have a look at them and respond to them (feel free to disagree with anything by the way!), and then approve and merge after that. While going through, you'll see some resolved comments from me - that's just me pointing out things I'm super happy with 🌟
> 
> Since this is the first PR in this repository and what you've done is awesome, I'd love to share this on my LinkedIn (https://www.linkedin.com/in/christa-bridges/). Would you be happy for me to do that, and have an account so I can tag you?
> 
> Many thanks and well done once again!

Thanks! This was a really fun project to work on! I'm still pretty new to contributing on GitHub so this is all very exciting and a little confusing for me haha. But I'm having an awesome time. 😎

And yeah please feel free to share this on LinkedIn, mine is https://www.linkedin.com/in/heysanj/

File: image-viewer.py
Comment (cBridges851) [inline image-viewer.py:33]: `!= len(self.image_files) - 1` <- not sure what this is preventing? Might not be needed.

File: image-viewer.py
Comment (cBridges851) [inline image-viewer.py:112]: I debate the value of this comment - I feel like `Resize the images for display if they are bigger than 500 x 300` is represented in the code, but `keeping their aspect ratio` helps explain the _why_, which provides value.

Maybe a `# keeping their aspect ratio` comment could be above the if statements on line 114 instead?

**Comment by cBridges851** [comment]
> > I'm really happy with this PR! I loved having a play with this locally, I didn't come across any issues and I was delighted to see the new ability to select a folder of images, as well as the additional things of resizing, refreshing, exiting, and the considerations around if a folder is empty - you've clearly put lots of consideration into this, so a big thank you and well done!
> > I've put a few minor suggestions on this PR, I'll let you have a look at them and respond to them (feel free to disagree with anything by the way!), and then approve and merge after that. While going through, you'll see some resolved comments from me - that's just me pointing out things I'm super happy with 🌟
> > Since this is the first PR in this repository and what you've done is awesome, I'd love to share this on my LinkedIn (https://www.linkedin.com/in/christa-bridges/). Would you be happy for me to do that, and have an account so I can tag you?
> > Many thanks and well done once again!
> 
> Thanks! This was a really fun project to work on! I'm still pretty new to contributing on GitHub so this is all very exciting and a little confusing for me haha. But I'm having an awesome time. 😎
> 
> And yeah please feel free to share this on LinkedIn, mine is https://www.linkedin.com/in/heysanj/

I'm really glad to hear that you've enjoyed working on this project and contributing to open source! I can very much relate to what you say about contributing on GitHub, it is very strange to get to grips with, it's certainly not something I've mastered yet 😂! The fun is the most important thing!

I've left a couple more tiny comments, should be good to merge after those are sorted! It would also be great to hear your thoughts on this comment: https://github.com/cBridges851/tkinter-image-viewer/pull/8#discussion_r987206902

Thank you for your LinkedIn! Will put a post out when this is merged 😊

**Comment by cBridges851** [comment]
Happy with the new comment around aspect ratio, just 2 small unresolved comments left :)!

**Comment by heySanj** [comment]
> Happy with the new comment around aspect ratio, just 2 small unresolved suggestions/questions left :)! You're doing a great job!

Nice! Looks good to me so I've resolved them, but unsure if I did it correctly or not haha 😅. Is it fine that I resolved them?

**Comment by cBridges851** [comment]
I'm more than happy for you to resolve comments 😊!

I realised on this one (https://github.com/cBridges851/tkinter-image-viewer/pull/8/files#r990228804) I misread the code, so definitely no further changes on that one haha, sorry about that!  

With this question (https://github.com/cBridges851/tkinter-image-viewer/pull/8#discussion_r987206902), I don't think having an option to select individual files should be a blocker to this merging, since the scope of this PR is around selecting a folder. I'll create a separate issue for that, and you're more than welcome to pick that up if you wish :)

I'll merge this now, very happy, and will get a LinkedIn post out shortly 😊! Thanks again for all the work you've done on this, keep enjoying open source and Hacktoberfest! Feel free to let me know if I can ever be of any assistance 😊

**Review by cBridges851 — APPROVED** [review]

**Comment by cBridges851** [comment]
@heySanj I've just released a post on LinkedIn, your profile didn't come up when I tried to tag you sorry, but here's the post! https://www.linkedin.com/posts/christa-bridges_softwareengineering-opensource-hacktoberfest-activity-6984501548089253888-_fFx?utm_source=share&utm_medium=member_desktop

Thanks again!

**Pull request merged**
